src/components/CheckReadPermissions.tsx

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+
+import { useApp } from '../app/useApp';
+import Config from '../config/Config';
+import { getSubApp } from '../utils/Basename';
+import { NotAuthorizedPage } from './NotAuthorized';
+
+export const CheckReadPermissions: React.FunctionComponent = (props) => {
+    const { rbac } = useApp();
```

Comment:
Change to `useAppContext` instead.

`useApp` calls the server again while `useAppContext` yields the already computed value.